torch/csrc/distributed/c10d/Hooks.cpp

```diff
@@ -15,44 +15,22 @@ namespace c10d {
 
 namespace {
 
-std::atomic<bool> event_queue_enabled = false;
-int sync_pipe;
-std::mutex event_queue_lock;
-std::deque<details::EventInfo> event_queue;
-
+std::mutex callbacks_lock;
+std::vector<CollectiveEventCallback> callbacks_list;
 } // namespace
 
-void enable_event_collection(int pipe) {
-  sync_pipe = pipe;
-  event_queue_enabled.store(true);
+TORCH_API void register_collective_callback(
+    CollectiveEventCallback&& callback) {
+  std::unique_lock<std::mutex> lock(callbacks_lock);
+  callbacks_list.push_back(std::move(callback));
 }
 
 namespace details {
 
-// we start dropping events after this
-const size_t MAX_QUEUE_SIZE = 512;
-
-bool dequeue_c10d_event(EventInfo& evt) {
-  std::unique_lock<std::mutex> lock(event_queue_lock);
-  if (event_queue.size() == 0) {
-    return false;
-  }
-  evt = event_queue.front();
-  event_queue.pop_front();
-  return true;
-}
-
 void enqueue_c10d_event(EventInfo&& evt) {
-  if (!event_queue_enabled.load())
-    return;
-
-  std::unique_lock<std::mutex> lock(event_queue_lock);
-  if (event_queue.size() >= MAX_QUEUE_SIZE) {
-    event_queue.back().drop_count++;
-  } else {
-    event_queue.push_back(std::move(evt));
-    char m = 'x';
-    write(sync_pipe, &m, 1);
+  std::unique_lock<std::mutex> lock(callbacks_lock);
+  for (auto& cb : callbacks_list) {
```

Comment:
nit: can we just use the full name `callback`? the name `cb` could mean thousand things.. Not strong opinion but just my thought. But since the program is short, should be fine.

Reply:
yep. ill do it in a follow up PR. i have a list of cleanups from the earlier PR too.